src/utils/StorageUtils.ts

```diff
@@ -0,0 +1,62 @@
+export type ValidStorageProviders = 'local' | 'session';
+
+export class StorageUtils<T> {
+  private storage: Storage;
+
+  constructor(public id: string, public storageProvider: ValidStorageProviders = 'local') {
+    // This check must be made in a try/catch. If cookies are disabled for a
+    // browser then window.localStorage will throw an undefined exception.
+    try {
+      switch (storageProvider) {
+        case 'local':
+          this.storage = window.localStorage;
+          break;
+        case 'session':
+          this.storage = window.sessionStorage;
+          break;
+        default:
+          this.storage = null;
+      }
+    } catch (error) {
+      this.storage = null;
+    }
+  }
+
+  public save(data: T) {
+    try {
+      if (this.storage != null) {
+        this.storage.setItem(this.getStorageKey(), JSON.stringify(data));
+      }
+    } catch (error) {}
+  }
+
+  public load(): T {
+    try {
+      if (this.storage == null) {
+        return null;
+      }
+      var value = this.storage.getItem(this.getStorageKey());
+      return value && JSON.parse(value);
+    } catch (error) {
+      return null;
+    }
+  }
+
+  public remove(key?: string) {
+    try {
+      if (this.storage != null) {
```

Comment:
Could you do an early return here instead of wrapping the block with an if?

Reply:
```
if (!this.storage) {
 return;
}

Reply:
At first I left the code almost the same as LocalStorageUtils to avoid any unexpected behaviour if we switch to this implementation. But it's ok, makes sense.